### PR TITLE
Add possibility to book days

### LINKED ISSUE #95: Use days instead of minutes as timeslots
**Description**
The current implementation only allows timeslots as minutes, not as days. Using days instead of minutes should be an additional feature.

**Links/References**
- Times do not need to be set, only days
- Days could be selected as blacklist or whitelist

**Tasks**

- [ ] Add a migration for the database (#48) to allow setting days
- [ ] Add the possibility to set a blacklist (or whitelist) for the days in the backend
- [ ] Add the possibility to set them in `/owner`
- [ ] Show the availability of the days in `/available` (maybe a general fix for #12 might make sense here)
- [ ] Add a frontend to select the days in a calendar

**Additional Context**
Using days makes more sense in a setting like camping spaces for example.

### PR DESCRIPTION
**Pull request by Narigo**
**Checklist**

- [x] Resolves #95
- [x] The [code of conduct](../blob/master/.github/CODE_OF_CONDUCT.md) is respected
- [x] Added project
- [x] Added label "Review needed"
- [x] Assigned a reviewer

**Reviewer**

- [ ] I've checked out the code and tested it myself.

**Changes**

- Added possibility to create shops with day slots
- Added automated test framework
- Added frontend for selecting day ranges
- Added possibility to set holidays when shop is in "weekly opening hour" mode
- Using "00:00:00" as start and end times should result in "whole day open" for "weekly opening hour" mode
- Added possibility to migrate database schemas
- Updated packages (and fixed breaking changes)

**Notes**

Sorry for the way too big PR. I was way too much in research mode and trial/error until I came to a good conclusion. I should have split this PR up into multiple parts instead...

*Edit:* Actually, there was a part of it split up before: https://github.com/compose-us-research/noqueue/pull/96 - this PR is based on that pull request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,34 +37,58 @@ Die Basisversion stellen wir kostenfrei zur Verfügung. Wem die WebApp hilft, de
...
+3. `docker-compose -f dev-docker-compose.yml run --service-ports --rm frontend bash` -
+   runs the frontend container: In the shell, have a look into `/app/frontend`
+   and run one of the frontend scripts after installing the dependencies
+   (`yarn`). `yarn storybook` starts Storybook and allows you to change the
+   various components with hot module reload. `yarn start` starts a development
+   server to see changes live in the app. Since the live backend will send you
+   the URL of the backend service when selecting a shop, the URL might change
+   and as a result changes might not be visible. Use the local URL to check
+   this. With `yarn build`, a static version of the page will be built and the
+   live backend is going to use that.
```
Comment (cowglow) [inline README.md:94]: Yarn was dropped, Please update text to use NPM.

**Review by cowglow — COMMENTED** [review]
Did a quick scan and it looks good.

**Review by cowglow — CHANGES_REQUESTED** [review]
Able to merge after changes are made

**Comment by Narigo** [comment]
Updated the Readme files, will merge now

**Pull request merged**